pkg/build/builder/source.go

```diff
@@ -256,16 +268,43 @@ func copyImageSource(dockerClient DockerClient, containerID, sourceDir, destDir
 	return tarHelper.ExtractTarStreamWithLogging(destDir, file, tarOutput)
 }
 
-func extractSourceFromImage(dockerClient DockerClient, image, buildDir string, paths []api.ImageSourcePath) error {
+func extractSourceFromImage(dockerClient DockerClient, image, buildDir string, imageSecretIndex int, paths []api.ImageSourcePath, forcePull bool) error {
 	glog.V(4).Infof("Extracting image source from %s", image)
 
-	// Pre-pull image if a secret is specified
-	pullSecret := os.Getenv(dockercfg.PullSourceAuthType)
-	if len(pullSecret) > 0 {
-		dockerAuth, present := dockercfg.NewHelper().GetDockerAuth(image, dockercfg.PullSourceAuthType)
-		if present {
-			dockerClient.PullImage(docker.PullImageOptions{Repository: image}, dockerAuth)
+	dockerAuth := docker.AuthConfiguration{}
+	if imageSecretIndex != -1 {
+		pullSecret := os.Getenv(fmt.Sprintf("%s%d", dockercfg.PullSourceAuthType, imageSecretIndex))
```

Comment:
would be handy to look at the specific envvar (`PULL_SOURCE_DOCKERCFG_PATH_<N>`), then fall back to a general envvar (`PULL_SOURCE_DOCKERCFG_PATH`) if unspecified... to avoid needing to set the same value into N envvars if you had N source images coming from the same registry